roles/acme/tasks/challenge/dns-01/hetzner.yml

```diff
@@ -1,28 +1,29 @@
 ---
 ### include/role 3 - validate challenge
-- name: Validate challenge only if it is created or changed  # noqa no-handler
+- name: Validate challenge only if it is created or changed # noqa no-handler
   when: challenge is changed
   block:
     - name: Lookup the zone_id using the acme_domain.zone
       ansible.builtin.uri:
-        url: "https://dns.hetzner.com/api/v1/zones?name={{ acme_domain.zone }}"
+        url: https://dns.hetzner.com/api/v1/zones?name={{ acme_domain.zone }}
         headers:
           Auth-API-Token: "{{ acme_hetzner_auth_token }}"
       register: lookup_zone_id
 
     - name: Add a new TXT record to the SAN domains
       ansible.builtin.uri:
-        url: "https://dns.hetzner.com/api/v1/records"
+        url: https://dns.hetzner.com/api/v1/records
         method: POST
         body_format: json
         body:
           {
-            "name": "{{ '_acme-challenge' }}{{ '' if item | replace('*.', '') == (item.split('.')[-2:] | join('.')) else '.' }}{{ (item | replace('*.', '')).split('.')[:-2] | join('.') }}",
+            "name": "{{ '_acme-challenge' }}{{ '' if item | replace('*.', '') == (item.split('.')[-2:] | join('.')) else '.' }}\
```

Comment:
why is the `'_acme_challenge'` string wrapped in curly braces?
Also this construct sems like a very good target for simplification!
Is there an explaination, why this is so complacated compared to the other DNS implementations?

(This need not be solved in this PR)

Reply:
the git log says:

> I had to make some adjustments to get this to work for Hetzner DNS. To add a record you need a zone ID, and the acme challenge TXT records need to include the sub domain. If you have a wildcard this needs to be stripped also. Similar to your azure module.

I'm not sure how to proceed here. To simplify this, we'd need some good tests to verify that it still works. And before I touch this peace of code, I'll replace it with a task from the hetzner dns module.. https://github.com/T-Systems-MMS/ansible-collection-acme/issues/62